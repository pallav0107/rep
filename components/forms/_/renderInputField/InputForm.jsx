import React from 'react';
import PropTypes from 'prop-types';
import { Input } from '@wdpui/react-gel';
import {
  ErrorIcon,
  InputErrorWrapper,
  ErrorText,
} from '../../../../style/style';

export const determineValidation = (touched, error) => {
  if (touched && error) return 'danger';
  return null;
};

const InputForm = ({ input, type, multiline,rows, placeholder,meta: { touched, error } }) => (
  <span>
    {(type === 'text' ||
      type === 'email' ||
      type === 'number' ||
      type === 'password' ||
      type === 'Date') && (
      <Input {...input} type={type} placeholder={placeholder} multiline={multiline} rows={rows}/>
    )}
    {touched &&
      (error && (
        <InputErrorWrapper>
          <ErrorIcon size="xsmall" />
          <ErrorText styling="danger">{error}</ErrorText>
        </InputErrorWrapper>
      ))}
  </span>
);

InputForm.propTypes = {
  input: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired,
  placeholder: PropTypes.string,
};

InputForm.defaultProps = {
  input: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired,
  placeholder: null,
};

export default InputForm;
