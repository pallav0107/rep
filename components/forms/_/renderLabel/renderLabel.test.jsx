import React from 'react';
import { shallow } from 'enzyme';
import InputForm from './renderLabel';
import "../../../../setupTests"

describe('RenderLabel', () => {
  it('should render', () => {
    const component = <InputForm />;
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
