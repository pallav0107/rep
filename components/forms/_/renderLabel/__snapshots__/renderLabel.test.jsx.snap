// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderLabel should render 1`] = `<span />`;
